File: cfgs/java_cfgs_foo/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP.java
Nodes
1;special;;;
2;conditional;foo_1 > ((foo_2 - 2) / 2);14;
3;statement;return true;15;
4;exit;;;
5;conditional;(((foo_1[foo_2] >= foo_3[(2 * foo_4) + 1]) && (foo_5[foo_6] >= foo_7[(2 * foo_8) + 2])) && fun_1(foo_9, (2 * foo_10) + 1, foo_11)) && fun_2(foo_12, (2 * foo_13) + 2, foo_14);17;
6;statement;return true;18;
7;statement;return false;20;
Edges
0;2;
1;3;5;
2;4;
3;
4;6;7;
5;4;
6;4;

File: cfgs/pythoncfgs/HOW_TO_CHECK_IF_A_GIVEN_ARRAY_REPRESENTS_A_BINARY_HEAP.py
Nodes
1;conditional;foo_1 > fun_1((foo_2 - 2) / 2);8;
2;conditional;foo_1[foo_2] >= foo_3[2 * foo_4 + 1] and foo_5[foo_6] >= foo_7[2 * foo_8 + 2] and fun_1(foo_9, 2 * foo_10 + 1, foo_11) and fun_2(foo_12, 2 * foo_13 + 2, foo_14);10;
3;statement;return false;12;
4;statement;return true;11;
5;statement;return true;9;
6;exit;;
Edges
0;2;5;
1;3;4;
2;6;
3;6;
4;6;
5;